src/components/PostInfo/PostInfo.tsx

```diff
@@ -1,5 +1,35 @@
 import React from 'react';
+import './PostInfo.scss';
+import { Posts } from '../../types/Posts';
+import { CommentList } from '../CommentList';
+import { UserInfo } from '../UserInfo';
 
-export const PostInfo: React.FC = () => (
-  <>Put the post here</>
-);
+type Props = {
+  post: Posts,
+};
+
+export const PostInfo: React.FC<Props> = ({ post }) => {
+  return (
+    <div className="PostInfo">
+      <div className="PostInfo__header">
+        <h3 className="PostInfo__title">{post.title}</h3>
+
+        <p>
+          {' Posted by  '}
+
+          {post.user && <UserInfo user={post.user} />}
+        </p>
+      </div>
+
+      <p className="PostInfo__body">
+        {post.body}
+      </p>
+
+      <hr />
+
+      {post.comments?.length !== 0
+        ? <CommentList comments={post.comments} />
```

Comment:
Definitely not a mistake here, but as a rule of thumb it's recommended to extract that kind of logic into a separate function. 

Like, 

```
{renderComments()}


const renderComments = () => {
  if(!posts.comments?.length) return ( <b data-cy="NoCommentsMessage">No comments yet</b> )
  
  return (<CommentList comments={post.comments} />)
}
```

It will just make the code more readable and easier to understand in case you have some really complex logic or lot's of exceptions.